arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -758,17 +773,27 @@ public fun <A, B> Iterable<A>.align(b: Iterable<B>): List<Ior<A, B>> =
   this.align(b, ::identity)
 
 /**
- * aligns two structures and combine them with the given [Semigroup.combine]
+ * aligns two structures and combine them with [combine]
  */
 public fun <A> Iterable<A>.salign(
-  SG: Semigroup<A>,
+  combine: (A, A) -> A,
   other: Iterable<A>
-): Iterable<A> = SG.run {
+): Iterable<A> =
   align(other) {
     it.fold(::identity, ::identity) { a, b ->
-      a.combine(b)
+      combine(a, b)
     }
   }
+
+/**
+ * aligns two structures and combine them with the given [Semigroup.combine]
+ */
+@Deprecated(SemigroupDeprecation, ReplaceWith("SG.run { salign({ a, b -> a.combine(b) }, other) }"))
+public fun <A> Iterable<A>.salign(
```

Comment:
can we make this internal or this exists to not break binaries?

Reply:
This is the currently released API, which we're deprecating in favour of the new `(A, A) -> A` API.